app/src/test/java/com/codesoom/assignment/application/TaskServiceTest.java

```diff
@@ -0,0 +1,217 @@
+                givenTasks.add(withIdAndTitle(1L, "할 일 추가1"));
+                givenTasks.add(withIdAndTitle(2L, "할 일 추가2"));
+
+                taskService.createTask(givenTasks.get(0));
+                taskService.createTask(givenTasks.get(1));
+
+            }
+
+            @Test
+            @DisplayName("모든 할 일을 리턴한다")
+            void it_returns_all_tasks() {
+                List<Task> findTasks = subject();
+                final int actualSize = findTasks.size();
+
+                assertThat(findTasks).isEqualTo(givenTasks);
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("createTask 메소드는")
+    class Describe_createTask {
+        @Nested
+        @DisplayName("새로운 할 일이 주어지면")
+        class Context_with_new_task extends TaskFactory {
+            private final Long TASK_ID = 1L;
+            private final String TITLE = "할 일 추가";
+            private final Task givenTask = withTitle(TITLE);
+
+            @Test
+            @DisplayName("DB에 등록하고 등록된 할 일을 리턴한다")
+            void it_returns_registered_task() {
+                final int expectedSize = taskService.getTasks().size() + 1;
+
+                Task actualTask = taskService.createTask(givenTask);
+
+                final int actualSize = taskService.getTasks().size();
+                final Task actualFindTask = taskService.getTask(TASK_ID);
+
+                assertThat(actualSize).isEqualTo(expectedSize);
+                assertThat(actualTask.getTitle()).isEqualTo(givenTask.getTitle());
+                assertThat(actualFindTask.getTitle()).isEqualTo(givenTask.getTitle());
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("getTask 메소드는")
+    class Describe_getTask {
+        @Nested
+        @DisplayName("ID가 주어지면")
+        class Context_with_existing_task_id extends TaskFactory {
+            private final Long TASK_ID = 1L;
+            private final String TITLE = "할 일 검색";
+            private Task givenTask;
+
+            @BeforeEach
+            void prepareTests() {
+                givenTask = withTitle(TITLE);
+                taskService.createTask(givenTask);
+            }
+
+            @Test
+            @DisplayName("할 일을 찾아 리턴한다")
+            void it_returns_found_task() {
+                Task actualTask = taskService.getTask(TASK_ID);
+
+                assertThat(actualTask.getTitle()).isEqualTo(givenTask.getTitle());
+            }
+
+        }
+
+        @Nested
+        @DisplayName("유효하지 않은 ID가 주어지면")
+        class Context_with_non_existing_task_id {
+            @Test
+            @DisplayName("예외를 던진다")
+            void it_throws_task_not_found_exception() {
+                assertThatThrownBy(() -> taskService.getTask(100L)).isInstanceOf(TaskNotFoundException.class);
+            }
+        }
+
+    }
+
+    @Nested
+    @DisplayName("updateTask 메소드는")
+    class Describe_updateTask {
+        @Nested
+        @DisplayName("ID가 주어지면")
+        class Context_with_existing_task_id extends TaskFactory {
```

Comment:
`extends TaskFactory`는 불필요해 보입니다.

```suggestion
        class Context_with_existing_task_id {
```

Reply:
굳이 상속을 사용할 필요는 없었는데 상속을 통한 테스트 중복제거를 한번 사용해보고싶어서 적용해봤습니다.